jni/CMakeLists.txt

```diff
@@ -129,7 +129,7 @@ if (${CONFIG_FAISS} STREQUAL ON OR ${CONFIG_ALL} STREQUAL ON OR ${CONFIG_TEST} S
         set(SIMD_ENABLED true)   # set default value as true if the argument is not set
     endif()
 
-    if(${CMAKE_SYSTEM_NAME} STREQUAL Windows OR ${CMAKE_SYSTEM_PROCESSOR} MATCHES "aarch64" OR NOT ${SIMD_ENABLED})
+    if(${CMAKE_SYSTEM_NAME} STREQUAL Windows OR ${CMAKE_SYSTEM_PROCESSOR} MATCHES "aarch64" OR ${CMAKE_SYSTEM_PROCESSOR} MATCHES "arm64" OR NOT ${SIMD_ENABLED})
```

Comment:
if that's a separate fix unrelated to the logic on this feature I'd rather put into a separate PR mainly because it's easy to track and backport and it can be merged earlier to unblock other PRs (as this PR may take some time to merge)